### ISSUE TITLE
Sitewide, design system [AXE-CORE]: Heading levels SHOULD increase by one

### ISSUE THREAD

**Issue by jenstrickland** [department-of-veterans-affairs/va.gov-team#7244]
# [508-defect-2](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-2)

**Feedback framework**
- **❗️ Must** for if the feedback must be applied 
- **⚠️Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Description

When search criteria includes Urgent care for facility type and Community urgent care providers for service type an alert box appears. The "Important information about your Community Care appointment" heading is an `<h3>`, but there isn't an `<h2>` between it and the `<h1>`. 

Here is the content that is read to a screen reader user when they navigate by headings, where the screen reader will read, "heading level 1, Find VA Locations," followed by "heading level 3, Important information about…":
![IMG_6701](https://user-images.githubusercontent.com/57469/77690591-edd35f00-6f79-11ea-979b-2f854ce7f17c.jpg)

### Why it Matters
The underlying purpose of headers is to convey the structure of the page. For sighted users, the same purpose is achieved using different sizes of text. Text size, however, is not helpful for users of screen readers, because a screen reader identifies a header only if it is properly marked-up. When heading elements are applied correctly, the page becomes much easier to navigate for screen reader users and sighted users alike.

In the same way that sighted users can glance at a page and get a sense of its contents, users of screen readers can do the same by navigating through headings. Well written and properly ordered headings can save users, especially those who use screen readers, a lot of time and frustration.

The purpose of headings is to describe the structure of the webpage, not just highlight important text. They should be brief, clear, unique, and marked with h1 through h6 elements applied in hierarchical order. All of these qualities make headings valuable tools for screen reader users. Similar to the way sighted users can glance at a page and get a sense of its contents, screen reader users can navigate through headings. Well written and properly ordered headings can save screen reader time and frustration.

In addition to making the page more accessible, headings have other benefits since search engines use headings when filtering, ordering, and displaying results. Improving the accessibility of your site can also have the effect of making your page more findable.

## Point of Contact

**VFS Point of Contact:** Jennifer

## Acceptance Criteria

As a screen reader user, I want to navigate the hierarchy of the page content using heading levels to save time and frustration.

## Environment

* Operating System: all
* Browser: all
* Screenreading device: any
* Server destination: staging & production

## Steps to Recreate

1. Enter `https://staging.va.gov/find-locations` in browser
1. Have developer tools open, and the axe browser extension loaded
1. Enter Urgent care for facility type and Community urgent care providers for service type 
1. Run an axe audit
1. Verify that `<h3 class="usa-alert-heading" tabindex="-1">Important information about your Community Care appointment</h3>` is called out as an error of "Heading levels should only increase by one"

## Possible Fixes 

Change the structure of alert boxes to either:
1. use a `<dl>` with `role="alert"` to markup so it won't matter where they appear
1. use an `<h2>` for the heading, but this could be problematic if an alert appears in a section that is under an `<h3>`

## WCAG or Vendor Guidance (optional)

* [axe-core 3.4 - Heading levels should only increase by one](https://dequeuniversity.com/rules/axe/3.4/heading-order)
* [MDN dialog element](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/dialog)
* [MDN ARIA: alert role](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/Alert_Role)
* [MDN Description List element](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/dl)

## Screenshots or Trace Logs

![Screen Shot 2020-03-26 at 12 43 59 PM](https://user-images.githubusercontent.com/57469/77672628-801a3980-6f5f-11ea-9dc0-57617a37bcdb.png)

**Comment by mmiddaugh** [comment]
@jenstrickland this is ready for your validation, when you are able

**Comment by jenstrickland** [comment]
Validated. 🎉
